ports/stm32/powerctrl.c

```diff
@@ -260,10 +263,23 @@ int powerctrl_set_sysclk(uint32_t sysclk, uint32_t ahb, uint32_t apb1, uint32_t
 #endif
 
 void powerctrl_enter_stop_mode(void) {
+    #if MICROPY_HW_ENABLE_CYW43
+    cyw43_allow_sleep(&cyw43_state);
+    #endif
+
     // Disable IRQs so that the IRQ that wakes the device from stop mode is not
     // executed until after the clocks are reconfigured
     uint32_t irq_state = disable_irq();
 
+    #if defined(MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2)
+    mp_spiflash_deepsleep(&spi_bdev.spiflash, 1);
+    mp_spiflash_deepsleep(&spi_bdev2.spiflash, 1);
```

Comment:
This assumes `spi_bdev2` exists so will break on older board definitions that only have one spi flash defined

Reply:
Yes.  This needs to be configured better, at the moment it's just there so it works.

Reply:
👍 just a minor thing I saw, thought was worth noting as reminder for later! Looking fantastic in general though!

Reply:
Yeah, thanks.  There are a few other things like this that need cleaning up, eg the fixes to QSPI and the MPU.